pycsob/client.py

```diff
@@ -17,6 +20,191 @@ def send(self, request, **kwargs):
         return super(HTTPAdapter, self).send(request, **kwargs)
 
 
+class ConvertMixin:
+    """Convert instance into ordered dict."""
+
+    def to_dict(self) -> Dict[str, Union[str, Dict]]:
+        data = []
+        for field in fields(self):
+            value = getattr(self, field.name)
+            if value in conf.EMPTY_VALUES:
+                continue
+            if hasattr(value, 'to_dict'):
+                value = value.to_dict()
+            data.append((field.name, getattr(self, f"_format_{field.name}", lambda v: v)(value)))
+        return OrderedDict(data)
+
+
+@dataclass
+class CartItem(ConvertMixin):
+    """Cart item for creating card payment."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Basic-methods#item---cart-item-object-cart-
+
+    name: str
+    quantity: int
+    amount: int
+    description: Optional[str] = None
+    
+    def _format_name(self, value: str) -> str:
+        return value[:20].rstrip()
+    
+    def _format_description(self, value: str) -> str:
+        return value[:40].rstrip()
+
+
+@dataclass
+class CustomerAccount(ConvertMixin):
+    """Customer account data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customeraccount-data-
+
+    createdAt: Optional[str] = None
+    changedAt: Optional[str] = None
+    changedPwdAt: Optional[str] = None
+    orderHistory: Optional[int] = None
+    paymentsDay: Optional[int] = None
+    paymentsYear: Optional[int] = None
+    oneclickAdds: Optional[int] = None
+    suspicious: Optional[bool] = None
+
+
+@unique
+class CustomerLoginType(Enum):
+    """Type of customer login."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    GUEST = "guest"
+    ACCOUNT = "account"
+    FEDERATED = "federated"
+    ISSUER = "issuer"
+    THIRDPARTY = "thirdparty"
+    FIDO = "fido"
+    FIDO_SIGNED = "fido_signed"
+    API = "api"
+
+
+@dataclass
+class CustomerLogin(ConvertMixin):
+    """Customer login data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    auth: Optional[CustomerLoginType] = None
+    authAt: Optional[str] = None
+    authData: Optional[str] = None
+
+    def _format_auth(self, value: CustomerLoginType) -> str:
+        return value.value
+
+
+@dataclass
+class CustomerData(ConvertMixin):
```

Comment:
The class should just be named only `Customer`. In the documentation there is such a name. The word "data" is given outside this name.